### ISSUE TITLE
Problems building on OSX

### ISSUE THREAD

**Issue by peterfhannon** [mabrowning/minecrift#150]
Hi there, first of all, thank you for developing this, I'm really exctied about seeing it in action :)

I've had a few attempts at building minecrift for OSX, following the instructions, but unfortunately I am unable to get them to work. 

The first problem is that the configuration file for the minecrift install specifies minecraft version 1.8.1, however the MCP910Pre package that the instructions tell me to get and put into the project is configured to use minecraft 1.8.

I am able to change one or the other configs to make them match and download the appropriate minecraft jar, but either way I then run into problems that I am having a hard time solving.

A) If I configure minecrift install and MCP to use 1.8.1, the install process gets as far as applying FernFlower fixes and then hangs indefinitely. Debugging the python, I can see it hangs when calling *abstract_match* in **ffix.py** when processing **ChunkRenderDispatcher.java**.

B) If I change both to use 1.8, the install process gets as far as trying to decompile and then can't find the **minecraft_ff_in.jar**. This file exists, but it's not where the install process is expecting it. If I move it to where it should be manually, the process gets further but then fails with multiple errors when it attempts to apply patches. I figured I'd stop here and ask for some help.

Has anyone successfully built minecrift on OSX? Can anyone help at all please? 

The last commit of the version I am using is : **94ee26c5acc1cf10f52e2a879feec92731ec2e40**. (Is there a more convenient way to indicate what version of the codebase I'm on?)

**Comment by JesusFreke** [comment]
I was trying to build the 1.8.1 stuff (e.g. the "master" branch) last night on linux, and ran into exactly the same problems.

**Comment by StellaArtois** [comment]
If you get the latest from the 1.8.1 GitHub branch from scratch, copy in the mcp910pre1.zip to the root and run 

./install.sh

the Python script should patch MCP to work with 1.8.1, and patch the ChunkRenderDispatcher issue. Some initial errors during the MCP build are normal. The script will then patch the source with compile fixes, and then separately patch in the minecrift changes. Once the install script has finished, run 

./build.sh 

This will compile, build and create the installer archives without error.

Last time I checked this was all building ok on OSX?

**Comment by StellaArtois** [comment]
I'll check the master branch again as soon as possible. I'll also ensure the build instructions are up to date. 

Basically this should work:

Clean git checkout
Copy in mcp910pre1.zip to minecrift root
cd minecrift root
./install.sh
./build.sh

**Comment by JesusFreke** [comment]
hmm, I didn't copy the MCP zip to the minecrift root, but I did unzip it into the minecrift/mcp910-pre1 directory.

I didn't see anything in the installer.py about unzipping MCP itself, but I could easily have missed it.

**Comment by StellaArtois** [comment]
Just copy it in. Maybe the script fails to patch MCP otherwise. Again, I'll
make sure the instructions are clear on this and make the script a little
more robust if indeed its failing silently.

**Comment by JesusFreke** [comment]
@StellaArtois almost, but not quite :)


If I copy the mcp zip file into the minecrift directory, it doesn't appear to unzip it.
[Log](https://gist.github.com/JesusFreke/7478968957eb8d8f6837)

If I manually unzip the mcp zip into the correct directory, it makes a bit more progress:
[Log](https://gist.github.com/JesusFreke/1c644472bc9ced8b8541)

If I manually unzip the mcp and then update the client/server version in version.cfg, it gets a bit further.
[Log](https://gist.github.com/JesusFreke/45c2e2350871b583f84e)

Looking through install.py, it only unzips MCP if you specify the -c/--clean option. This appears to be the secret sauce that I needed.
https://gist.github.com/JesusFreke/f284dbd5f9fd3662368c

I haven't actually tried running it yet, but it appears to be building at least.

**Comment by JesusFreke** [comment]
Ok, I was eventually able to get it running, although head tracking does not seem to be working, unfortunately. I think you mentioned in another issue that you're aware of that particular issue and will get it fixed.

The 1 snag I ran into is that I wasn't able to use the prebuilt jrift library. I don't have the exact error handy, but it was some linking issue with libg++ or something. After I rebuilt jrift for my machine, I was able to get it running. Just without head tracking, so it's not very useful atm :)

**Comment by StellaArtois** [comment]
Ok, thanks for the investigation!

The scripts need a cleanup, sorry about that. I'll get those fixed. I think
that unzip issue has already been fixed in the 1.7.10 branch.

What version / distro of Linux are you using?

**Comment by JesusFreke** [comment]
ubuntu 14.04.2, 64-bit

**Comment by StellaArtois** [comment]
Built the libs on Ubuntu 14.10. I assume it rebuild fine using the JRift cmake script for you?

**Comment by JesusFreke** [comment]
Yep, I used the build_linux.sh script. The only "gotcha" was that I didn't realize that the OculusSDK directory was actually a submodule that I had to do the init+update dance for. And then once again, the cmake directory *within* OculusSDK.

**Comment by StellaArtois** [comment]
I’ll make sure the instructions are updated for this. I’ll add a sample script to the readme to get *everything* out from scratch and building.

Question: If I build JRift on 14.10, what can I do to ensure it will run on 14.04, and below? Are there some compiler flags that will allow me to knock down the 'API level' or similar?

**Comment by JesusFreke** [comment]
Yeah, I don't really know about that. sorry! :)

**Comment by peterfhannon** [comment]
Just to let you know I got it working thanks to the help above. Head tracking works for me (I'm on OSX Mavericks).
It is a pretty amazing experience, so thanks to everyone involved for creating this!

**Comment by StellaArtois** [comment]
I'll be porting over some script enhancements to the port_to_1.8.1 branch soon. The branch port_to_1.7.10 is currently the most up-to-date, with some updates to the README.md to help explain how to get the scripts running.
